src/libServer/IsolatedServer.cpp

```diff
@@ -491,6 +549,166 @@ Json::Value IsolatedServer::CreateTransaction(const Json::Value& _json) {
   }
 }
 
+Json::Value IsolatedServer::CreateTransactionEth(EthFields const& fields,
+                                                 bytes const& pubKey) {
+  try {
+    if (m_pause) {
+      throw JsonRpcException(RPC_INTERNAL_ERROR, "IsoServer is paused");
+    }
+
+    lock_guard<mutex> g(m_blockMutex);
```

Comment:
Nitpick: Do we need this lock here? Can we move it to the relevant scope it really protects?

Reply:
Again, I was following the code from add TX from what was there before. It looks like the mutex should be protecting changes to the account store (UpdateAccountsTemp) and the m_mediator.m_txBlockChain. 

I have moved the locks to the last line which looks safe (just before validateTX). The way the code is written, I think that if you lock any later there is the potential of multiple TXs passing the nonce check and proceeding to execute, although the code is hard to reason about so I am not totally sure.